crates/viewer/re_view_dataframe/src/display_record_batch.rs

```diff
@@ -182,23 +179,45 @@ impl DisplayColumn {
     #[allow(clippy::borrowed_box)] // https://github.com/rust-lang/rust-clippy/issues/11940
     fn try_new(
         column_descriptor: &ColumnDescriptor,
-        column_data: &Box<dyn Arrow2Array>,
+        column_data: &ArrowArrayRef,
     ) -> Result<Self, DisplayRecordBatchError> {
+        fn int64_from_nanoseconds(
+            duration_array: &ArrowTimestampNanosecondArray,
+        ) -> ArrowInt64Array {
+            let data = duration_array.to_data();
+            let buffer = data.buffers()[0].clone();
+            let int64_data = arrow::array::ArrayData::builder(arrow::datatypes::DataType::Int64)
+                .len(duration_array.len())
+                .add_buffer(buffer)
+                .build()
+                .expect("Nanonseconds should be represented as i64");
```

Comment:
Both of these statements risk crashing the entire viewer even in release. As unlikely as it is, it's still not worth losing user data over.
It should only crash in debug, and either return a zeroed array or empty array or `None` or an error in release.

Reply:
Good catch